pkg/plugin/util/util.go

```diff
@@ -297,3 +299,76 @@ func IsResourceBlockedOnRestore(resourceName string) bool {
 	return constants.ResourcesToBlockOnRestore[resourceName]
 }
 
+func GetPVForPVC(pvc *corev1.PersistentVolumeClaim, pvGetter corev1client.PersistentVolumesGetter) (*corev1.PersistentVolume, error) {
+	if pvc.Spec.VolumeName == "" {
+		return nil, errors.Errorf("PVC %s/%s has no volume backing this claim", pvc.Namespace, pvc.Name)
+	}
+	if pvc.Status.Phase != corev1.ClaimBound {
+		// TODO: confirm if this PVC should be snapshotted if it has no PV bound
+		return nil, errors.Errorf("PVC %s/%s is in phase %v and is not bound to a volume", pvc.Namespace, pvc.Name, pvc.Status.Phase)
+	}
+	pvName := pvc.Spec.VolumeName
+	pv, err := pvGetter.PersistentVolumes().Get(context.TODO(), pvName, metav1.GetOptions{})
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to get PV %s for PVC %s/%s", pvName, pvc.Namespace, pvc.Name)
+	}
+	return pv, nil
+}
+
+func Contains(slice []string, key string) bool {
+	for _, i := range slice {
+		if i == key {
+			return true
+		}
+	}
+	return false
+}
+
+func GetPodsUsingPVC(pvcNamespace, pvcName string, podsGetter corev1client.PodsGetter) ([]corev1.Pod, error) {
+	podsUsingPVC := []corev1.Pod{}
+	podList, err := podsGetter.Pods(pvcNamespace).List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	for _, p := range podList.Items {
+		for _, v := range p.Spec.Volumes {
```

Comment:
the pod could be shut down as well, you can refer to https://github.com/kubernetes/kubernetes/blob/29e4fdab4d644af51c1fa2623bf0e78f3dd6383b/pkg/controller/volume/pvcprotection/pvc_protection_controller.go#L295 for similar checks.

Reply:
Will leave it as a separate task.